types/row_proof.go

```diff
@@ -16,9 +16,17 @@ type RowProof struct {
 	RowRoots []tmbytes.HexBytes `json:"row_roots"`
 	// Proofs is a list of Merkle proofs where each proof proves that a row
 	// exists in a Merkle tree with a given data root.
-	Proofs   []*merkle.Proof `json:"proofs"`
-	StartRow uint32          `json:"start_row"`
-	EndRow   uint32          `json:"end_row"`
+	Proofs []*merkle.Proof `json:"proofs"`
```

Comment:
> Shouldn't that value be the same across all the proofs in RawProof.Proofs?
Yes
> Shouldn't that value indeed match the StartRow and EndRow of the RowProof?
Yes
> then perhaps the Proofs field is an ordered list of proofs based on their Index? 
Yes

Thing is, those two fields are redundant on the information that already exists in the proofs. I don't remember the reasoning behind adding them (probably was to support merkle range proofs at some point where we just have a set of side nodes, and a [start, end) range). 

But to avoid a breaking change, or having to remove those fields in a backwards compatible way, we decided to just add the comment. And if some teams downstream decide to use them, then they can validate them.

Also, if you prefer, we can the above validation. But since they're not used, it doesn't matter if we leave them the way they are or validate them.

Reply:
I see, then I'd defer it to you to whether that further checks on `StarRow` and `EndRow` or not.
However, I think it is still a valid check (and orthogonal to `StartRow` and `EndRow`) to see if the `Total` field is the same across all the proofs in `Proofs`.  Again, will leave the decision to you.

Reply:
cool, will add it in a separate PR. Thanks for signalling it 👍